src/marray.ts

```diff
@@ -61,6 +61,32 @@ export class MArray<T> {
     tu.insert(this.$data, 0, [t]);
   }
 
+  /**
+   * Like Array.slice - Returns a section of an array.
+   * @param start The beginning of the specified portion of the array.
+   * @param end The end of the specified portion of the array, exclusive of the element at the index 'end'.
+   */
+  slice(start?: number, end?: number) {
+    const extractedElements = new MArray<T>();
+
+    if (start > this.length) {
+      return extractedElements;
+    } else if (start < 0) {
+      start = this.length + start;
+    }
+    start = start ?? 0;
+
+    if (end < 0) {
+      end = this.length + end;
+    }
+    end = !end || end > this.length ? this.length : end;
+
+    for (let i = start; i < end; i++) {
```

Comment:
It's probably faster to use iterateData with index and count here as it wouldn't have to restart from the top of the tree on every index

Reply:
Like [this](https://github.com/hfour/yads/pull/25/commits/98cfc7213a288a9a4cf28a4d1b03f87e8db54910)?